sound/soc/sof/intel/hda-codec.c

```diff
@@ -39,54 +39,40 @@ static void hda_codec_load_module(struct hda_codec *codec) {}
 #endif
 
 /* check jack status after resuming from suspend mode */
-void hda_codec_jack_check(struct snd_sof_dev *sdev, int status)
+void hda_codec_jack_check(struct snd_sof_dev *sdev, u32 status)
 {
-	struct sof_intel_hda_dev *hda = sdev->pdata->hw_pdata;
 	struct hda_bus *hbus = sof_to_hbus(sdev);
 	struct hdac_bus *bus = sof_to_bus(sdev);
 	struct hda_codec *codec;
-	int mask;
-
-	/*
-	 * there are two reasons for runtime resume
-	 * (1) waken up by interrupt triggered by WAKEEN feature
-	 * (2) waken up by pm get functions for some audio operations
-	 * For case (1), the bits in status mean which codec triggers
-	 * the interrupt and jacks will be checked on these codecs.
-	 * For case (2), we need to check all the non-hdmi codecs for some
-	 * cases like playback with HDMI or capture with DMIC. In these
-	 * cases, only controller is active and codecs are suspended, so
-	 * codecs can't send unsolicited event to controller. The jack polling
-	 * operation will activate codecs and unsolicited event can work
-	 * even codecs become suspended later.
-	 */
-	mask = status ? status : hda->hda_codec_mask;
+
+	/* disable controller Wake Up event*/
+	snd_hdac_chip_writew(bus, WAKEEN,
+			     snd_hdac_chip_readw(bus, WAKEEN) &
+			     ~STATESTS_INT_MASK);
+
+	if (!status)
+		return;
 
 	list_for_each_codec(codec, hbus)
-		if (mask & BIT(codec->core.addr))
+		if (status & BIT(codec->core.addr) && codec->jacktbl.used) {
```

Comment:
this is a bit confusing to have both types of AND in the same test, I don't even know what precedence rules apply?